tools/scripts/start-local-registry.ts

```diff
@@ -48,63 +73,77 @@ function startLocalRegistry({
+          'nx',
+          ...`run ${localRegistryTarget} --location none --clear true`.split(
+            ' ',
+          ),
+          ...(storage ? [`--storage`, storage] : []),
+        ],
+        { stdio: 'pipe', shell: true, detached: true },
+      );
 
-    const listener = data => {
-      if (verbose) {
-        process.stdout.write(data);
-      }
-      if (data.toString().includes('http://localhost:')) {
-        const port = parseInt(
-          data.toString().match(/localhost:(?<port>\d+)/)?.groups?.port,
-        );
-        console.info('Local registry started on port ' + port);
+      const listener = data => {
+        if (verbose) {
+          process.stdout.write(data);
+        }
+        if (data.toString().includes('http://localhost:')) {
+          const port = parseInt(
+            data.toString().match(/localhost:(?<port>\d+)/)?.groups?.port,
+          );
+          console.info('Local registry started on port ' + port);
 
-        const registry = `http://localhost:${port}`;
-        process.env.npm_config_registry = registry;
-        execSync(
-          `npm config set //localhost:${port}/:_authToken "secretVerdaccioToken"`,
-        );
+          const registry = `http://localhost:${port}`;
+          process.env.npm_config_registry = registry;
+          execSync(
+            `npm config set //localhost:${port}/:_authToken "secretVerdaccioToken"`,
+          );
 
-        // yarnv1
-        process.env.YARN_REGISTRY = registry;
-        // yarnv2
-        process.env.YARN_NPM_REGISTRY_SERVER = registry;
-        process.env.YARN_UNSAFE_HTTP_WHITELIST = 'localhost';
+          // yarnv1
+          process.env.YARN_REGISTRY = registry;
+          // yarnv2
+          process.env.YARN_NPM_REGISTRY_SERVER = registry;
+          process.env.YARN_UNSAFE_HTTP_WHITELIST = 'localhost';
 
-        console.info('Set npm and yarn config registry to ' + registry);
+          console.info('Set npm and yarn config registry to ' + registry);
 
-        resolve(() => {
-          childProcess.kill();
-          execSync(`npm config delete //localhost:${port}/:_authToken`);
-        });
-        childProcess?.stdout?.off('data', listener);
-      }
-    };
-    childProcess?.stdout?.on('data', listener);
-    childProcess?.stderr?.on('data', data => {
-      process.stderr.write(data);
-    });
-    childProcess.on('error', err => {
-      console.error('local registry error', err);
-      reject(err);
-    });
-    childProcess.on('exit', code => {
-      console.info('local registry exit', code);
-      if (code !== 0) {
-        reject(code);
-      } else {
-        resolve(() => {});
-      }
-    });
-  });
+          resolve({
+            registry,
+            stop: () => {
+              if (childProcess.pid) {
+                process.kill(-childProcess.pid);
+              }
+              // does not kill the underlying process, see https://github.com/nodejs/node/issues/46865
+              childProcess.kill();
```

Comment:
this was the second culprit why the local registry was not stopped properly; this is related to a UNIX design ~~bug~~ feature.